blockchain/client.go

```diff
@@ -128,7 +128,7 @@ func (c *Client) ListenEvents(
 	})
 
 	// Sequence historical and live headers.
-	headersC := make(chan types.Header)
+	headersC := make(chan types.Header, 2)
```

Comment:
why 2?

Reply:
Buffer size defines how much previous operations we allow before the next one in pipeline consumes the result. Two looks just sane. Its not too much to not consume unnecessary memory or occupy network resources needed for the next operation. And its not too low - we can fetch a couple of historical headers ahead of time if an events listening callback works with a certain block for too long.

Reply:
> 1. wouldn't this async approach affects the consistency? e.g. we got block 4 and 5. we processed 5th block, it has been stored as known block but block 4 failed. or it won't be a case? or e.g. in block 5 we have bucketUpdatedEvent and it was processed before 4th block with bucketCreatedEvent.

This particular patch does not break consistency. The client side still receive blocks in sequentially, but we do our networking and pre-fetch historical headers and events while client-side code does its business.

> 2. might be it make sense to move it into client configuration? so we can experiment on the value..

It makes sense if we invoke client side functions (events listening callbacks) asynchronously and the client may want to have an ability to set the size of an "executors pool" for them. But here such a configuration does not change affect the process. The patch speeds up each block processing for a duration of an RPC call (about 1 second) even with 1 as a buffer size, but further extension of the buffer does not help with networking.